delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,301 @@
+                IO.raiseError(NoIdentities)
+              else {
+                val missing = setup.identities.diff(caller.identities)
+                if (missing.isEmpty) {
+                  IO.unit
+                } else {
+                  IO.raiseError(InvalidIdentities(missing))
+                }
+              }
+            }
+            .getOrElse(IO.raiseError(NoCrossProjectSetup))
+      }
+
+    }
+
+    (state, command) match {
+      /**
+        * Creating a resolver
+        */
+      // The resolver already exists
+      case (_: Current, c: CreateResolver)                                        =>
+        IO.raiseError(ResolverAlreadyExists(c.id, c.project))
+      // Create a resolver
+      case (Initial, c: CreateResolver)                                           =>
+        for {
+          _   <- validateCrossProjectSetup(c.`type`, c.crossProjectSetup)
+          now <- instant
+        } yield ResolverCreated(
+          id = c.id,
+          project = c.project,
+          `type` = c.`type`,
+          crossProjectSetup = c.crossProjectSetup,
+          priority = c.priority,
+          rev = 1L,
+          instant = now,
+          subject = caller.subject
+        )
+
+      /**
+        * Updating a resolver
+        */
+      // Update a non existing resolver
+      case (Initial, c: UpdateResolver)                                           =>
+        IO.raiseError(ResolverNotFound(c.id, c.project))
+      // Invalid revision has been provided
+      case (s: Current, c: UpdateResolver) if c.rev != s.rev                      =>
+        IO.raiseError(IncorrectRev(c.rev, s.rev))
+      // Resolver has been deprecated
+      case (s: Current, _: UpdateResolver) if s.deprecated                        =>
+        IO.raiseError(ResolverIsDeprecated(s.id))
+
+      // Update a resolver
+      case (s: Current, c: UpdateResolver)                                        =>
+        for {
+          _   <- if (s.`type` == c.`type`) IO.unit else IO.raiseError(DifferentResolverType(c.id, c.`type`, s.`type`))
+          _   <- validateCrossProjectSetup(c.`type`, c.crossProjectSetup)
+          now <- instant
+        } yield ResolverUpdated(
+          id = c.id,
+          project = c.project,
+          `type` = c.`type`,
+          crossProjectSetup = c.crossProjectSetup,
+          priority = c.priority,
+          rev = s.rev + 1,
+          instant = now,
+          subject = caller.subject
+        )
+
+      /**
+        * Tagging a resolver
+        */
+      // Resolver can't be found
+      case (Initial, c: TagResolver)                                              =>
+        IO.raiseError(ResolverNotFound(c.id, c.project))
+      // Invalid revision
+      case (s: Current, c: TagResolver) if c.rev != s.rev                         =>
+        IO.raiseError(IncorrectRev(c.rev, s.rev))
+      // Revision to tag is invalid
+      case (s: Current, c: TagResolver) if c.targetRev < 0 || c.targetRev > s.rev =>
```

Comment:
why do you need this `c.targetRev < 0` ? 
In this case, the state is `Current`, which in turn can never have rev =0. The only necessary check is `if c.targetRev >= s.rev`